frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,226 @@
+import * as React from 'react';
+import { Alert } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import {
+  Firehose,
+  FirehoseResource,
+  FirehoseResult,
+  LoadingBox,
+  HorizontalNav,
+  PageHeading,
+} from '@console/internal/components/utils';
+import { PodKind, k8sCreate, k8sKillByName } from '@console/internal/module/k8s';
+import { PodExecLoader } from '@console/internal/components/pod';
+import { PodModel } from '@console/internal/models';
+
+import { resourcePath } from './utils/resource-link';
+import { getBreadcrumbPath } from '@console/internal/components/utils/breadcrumbs';
+
+const getDebugPod = async (
+  debugName: string,
+  podToDebug: PodKind,
+  containerName: string,
+): Promise<PodKind> => {
+  const container = podToDebug?.spec?.containers?.find((c) => c.name === containerName);
+  return {
+    kind: 'Pod',
+    apiVersion: 'v1',
+    metadata: {
+      name: debugName,
+      namespace: podToDebug?.metadata?.namespace,
+      annotations: {
+        'debug.openshift.io/source-container': containerName,
+        'debug.openshift.io/source-resource': `/v1, Resource=pods/${podToDebug?.metadata?.name}`,
+      },
+    },
+    spec: {
+      activeDeadlineSeconds: 21600,
+      volumes: [],
+      containers: [
+        {
+          name: containerName,
+          image: container?.image,
+          command: ['/bin/sh'],
```

Comment:
It occurs to me that possibly more than just the command will be incorrect for a Windows pod.  @spadgett, do you know if Windows pods are different for debugging purposes?  If not, who should @zherman0 reach out to?

Reply:
Yeah, this would fail for Windows pods because of command, but also because you can't select a node. The CLI has a command-line flag for that. We might need to hide the button for windows pods for now and open a separate story.

Reply:
@zherman0 Here is the logic for we use for the normal pod terminal:

https://github.com/openshift/console/blob/master/frontend/public/components/pod-exec.jsx#L57-L60

We need to check if it's a Windows pod. If it is, we need to copy the node selector and tolerations, and we need to update the pod command to be `cmd`.

cc @aravindhp

Reply:
I added some windows logic, but I need to discuss more in detail what needs to be done.